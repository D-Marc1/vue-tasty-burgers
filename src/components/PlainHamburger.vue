<template>
  <button type="button" class="hamburger" :class="{'is-active' : isActive}" @click.prevent="toggle">
    <span class="hamburger-box">
      <span class="hamburger-inner">
        <span class="hamburger-inner__before" />
        <span class="hamburger-inner__after" />
      </span>
    </span>
  </button>
</template>

<script>
export default {
  props: {
    active: {
      type: Boolean,
      default: false
    }
  }
};
</script>

<style scoped>
.hamburger {
  padding: 0;
  display: inline-block;
  cursor: pointer;
  transition-property: opacity, filter;
  transition-duration: 0.15s;
  transition-timing-function: linear;
  font: inherit;
  color: inherit;
  text-transform: none;
  background-color: transparent !important;
  border: 0;
  margin: 0;
  overflow: visible;
}

.hamburger:hover,
.hamburger:focus {
  opacity: 0.7;
  outline: none;
}

.hamburger::-moz-focus-inner {
  border: 0;
}

.hamburger-box {
  width: 40px;
  height: 24px;
  display: block;
  position: relative;
  transform-origin: center left;
}

.hamburger-inner {
  display: block;
  top: 50%;
  margin-top: -2px;
}

.hamburger-inner,
.hamburger-inner__before,
.hamburger-inner__after {
  width: 40px;
  height: 4px;
  background-color: #000;
  border-radius: 4px;
  position: absolute;
  transition-property: transform;
  transition-duration: 0.15s;
  transition-timing-function: ease;
}

.hamburger-inner__before,
.hamburger-inner__after {
  content: "";
  display: block;
}

.hamburger-inner__before {
  top: -10px;
}

.hamburger-inner__after {
  bottom: -10px;
}
</style>