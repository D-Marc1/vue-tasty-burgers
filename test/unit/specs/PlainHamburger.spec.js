import { mount } from '@vue/test-utils'
import PlainHamburger from '@/src/components/PlainHamburger.vue'

const createCmp = propsData => mount(PlainHamburger, { propsData })

describe('PlainHamburger', () => {
  let wrapper

  beforeEach(() => {
    wrapper = createCmp()
  })

  it('is called', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('toggles the active status', () => {
    expect(wrapper.vm.isActive).toBe(false)

    wrapper.vm.toggle()
    expect(wrapper.vm.isActive).toBe(true)

    wrapper.vm.toggle()
    expect(wrapper.vm.isActive).toBe(false)

    wrapper.trigger('click')
    expect(wrapper.vm.isActive).toBe(true)

    wrapper.trigger('click')
    expect(wrapper.vm.isActive).toBe(false)
  })

  it('emits toggle event when prop changes', async () => {
    wrapper.setProps({ active: true })
    await wrapper.vm.$nextTick()
    expect(wrapper.emitted().toggle).toBeTruthy()
  })

  it('emits toggle event on click', async () => {
    wrapper.trigger('click')
    await wrapper.vm.$nextTick()
    expect(wrapper.emitted().toggle).toBeTruthy()
  })

  it('uses the right modifier class', () => {
    wrapper.setProps({ type: 'arrow' })
    expect(wrapper.vm.hamburgerModifierClass).toBe('hamburger--arrow')

    wrapper.setProps({ reversed: true })
    expect(wrapper.vm.hamburgerModifierClass).toBe('hamburger--arrow-r')

    wrapper.setProps({ type: 'arrow-r', reversed: false })
    expect(wrapper.vm.hamburgerModifierClass).toBe('hamburger--arrow-r')
  })
})
