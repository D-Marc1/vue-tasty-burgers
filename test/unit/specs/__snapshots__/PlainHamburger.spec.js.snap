// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlainHamburger renders correctly 1`] = `"<button type=\\"button\\" class=\\"hamburger hamburger--boring\\"><span class=\\"hamburger-box\\" style=\\"transform: scale(0.9); width: 36px; height: 23px;\\"><span class=\\"hamburger-inner\\" style=\\"background-color: rgb(0, 0, 0);\\"><span class=\\"hamburger-inner__before\\" style=\\"background-color: rgb(0, 0, 0);\\"></span> <span class=\\"hamburger-inner__after\\" style=\\"background-color: rgb(0, 0, 0);\\"></span></span></span></button>"`;
